### PR TITLE
Highlight the currently selected character

### PR DESCRIPTION
**Pull request by FamousJameous**
I thought it would be useful to highlight all occurrences of the number under the cursor as is done in many Sudoku apps. This makes it easier to find all occurrences of a certain number.

I am not a C or Curses expert so this commit is more of a suggestion for a more experienced developer to implement.

### CODE REVIEW

**Comment by jubalh** [comment]
Thanks for your contribution!

I have a couple of suggestions:
* let's use a different color for the highlighted characters vs the current cursor position (orange?)
* let's make this feature optional, pressing `M` (for marks) to enable it first

Would you be willing to implement it this way?

This feature is related to https://github.com/jubalh/nudoku/issues/7

**Comment by FamousJameous** [comment]
Thanks for the suggestions.

I definitely think making the feature optional is a good idea.

For the color, I want to make sure I understand. You want to have two highlight colors: one for the current cursor position (with white background and black foreground) and one for the other occurrences of the same number (with white background and orange foreground). I see that orange is not one of the eight basic colors, is 256 color support common enough to use it?

**Comment by FamousJameous** [comment]
I tried several color combinations and I liked magenta background with black foreground the best. It is visibly obvious without being distracting. What do you think?

**Comment by jubalh** [comment]
Looks good! Thanks for the changes!
I will look at the code tomorrow for more details :)

File: src/main.c [hunk 1]
```diff
@@ -510,10 +558,9 @@ int main(int argc, char *argv[])
 			if(plain_board[posy*9+posx] == '.')
 			{
 				// add inputted number to grid
-				wattron(grid, COLOR_PAIR(3));
-				wprintw(grid, "%c", key);
-				wattroff(grid, COLOR_PAIR(3));
 				user_board[posy*9+posx] = key;
+				// redraw grid to update highight
```
Comment (jubalh) [inline src/main.c]: small typo here `highight`

**Comment by jubalh** [comment]
Please try to change the git commit message.
See [here](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53)

Meaning no `.` in the first line. No more than 72 chars in the first line.

You can do this with `git rebase -i HEAD~3` or similar, and then edit the commit messages.

**Review by jubalh — CHANGES_REQUESTED** [review]

**Comment by FamousJameous** [comment]
I've never pushed rebased commits to Github before. It looks like it worked, but let me know if I messed something up.

**Comment by FamousJameous** [comment]
Whoops, sorry. I guess I forgot to build after all that rebase stuff. Should be fixed now.

**Comment by jubalh** [comment]
Very good job, thanks for your work.
I will add your name to the `AUTHORS` file, I will use the same format as for the others, if you don't want your email or real name or anything to be seen there just let me know.
If you are okay with it, let me know too :)

**Comment by FamousJameous** [comment]
I am trying to switch to a different email address than my Github default so use this if it works for you.
James 'FamousJameous' Warnock: James.Warnock.all@gmail.com

Thanks

**Comment by jubalh** [comment]
Done.

**Pull request merged**